crates/biome_css_semantic/src/semantic_model/specificity.rs

```diff
@@ -0,0 +1,180 @@
+    match name {
+        "where" => None,
+        "is" | "not" | "has" | "matches" => Some(ZERO_SPECIFICITY),
+        _ => Some(CLASS_SPECIFICITY),
+    }
+}
+
+fn evaluate_any_pseudo_class(class: &AnyCssPseudoClass) -> Specificity {
+    // https://www.w3.org/TR/selectors-4/#specificity-rules
+    match class {
+        AnyCssPseudoClass::CssBogusPseudoClass(_) => ZERO_SPECIFICITY,
+        AnyCssPseudoClass::CssPseudoClassFunctionCompoundSelector(selector) => {
+            CLASS_SPECIFICITY
+                + selector
+                    .selector()
+                    .map_or(ZERO_SPECIFICITY, |s| evaluate_any_compound_selector(&s))
+        }
+        AnyCssPseudoClass::CssPseudoClassFunctionCompoundSelectorList(selector_list) => {
+            let list_max = selector_list
+                .compound_selectors()
+                .into_iter()
+                .map(|s| s.map_or(ZERO_SPECIFICITY, |s| evaluate_any_compound_selector(&s)))
+                .reduce(|acc, e| acc.max(e))
+                .unwrap_or(ZERO_SPECIFICITY);
+
+            CLASS_SPECIFICITY + list_max
+        }
+        AnyCssPseudoClass::CssPseudoClassFunctionIdentifier(_) => CLASS_SPECIFICITY,
+        AnyCssPseudoClass::CssPseudoClassFunctionNth(_) => CLASS_SPECIFICITY,
+        AnyCssPseudoClass::CssPseudoClassFunctionRelativeSelectorList(selector_list) => {
+            if let Some(base) = selector_list
+                .name_token()
+                .ok()
+                .and_then(|name| evaluate_pseudo_function_selector(name.text()))
+            {
+                let list_max = selector_list
+                    .relative_selectors()
+                    .into_iter()
+                    .map(|relative_selector| {
+                        relative_selector
+                            .map_or(ZERO_SPECIFICITY, |s| evaluate_any_relative_selector(&s))
+                    })
+                    .reduce(|acc, e| acc.max(e))
+                    .unwrap_or(ZERO_SPECIFICITY);
+                base + list_max
+            } else {
+                ZERO_SPECIFICITY
+            }
+        }
+        AnyCssPseudoClass::CssPseudoClassFunctionSelector(s) => {
+            if let Some(base) = s
+                .name()
+                .ok()
+                .and_then(|name| evaluate_pseudo_function_selector(name.text()))
+            {
+                base + s.selector().map_or(ZERO_SPECIFICITY, |selector| {
+                    evaluate_any_selector(&selector)
+                })
+            } else {
+                ZERO_SPECIFICITY
+            }
+        }
+        AnyCssPseudoClass::CssPseudoClassFunctionSelectorList(selector_list) => {
+            if let Some(base) = selector_list
+                .name()
+                .ok()
+                .and_then(|name| evaluate_pseudo_function_selector(name.text()))
+            {
+                let list_max = selector_list
+                    .selectors()
+                    .into_iter()
```

Comment:
```suggestion
                    .iter()
```

Reply:
Fixed it https://github.com/biomejs/biome/pull/4097/commits/91f005cdc25cb42261acd326ed17dc3c58f5a397